hub/util/access_method.py

```diff
@@ -0,0 +1,99 @@
+import os
+import time
+import hub
+from hub.constants import TIMESTAMP_FILENAME
+from hub.util.exceptions import DatasetHandlerError
+from hub.util.storage import get_local_storage_path
+
+
+def check_access_method(access_method: str, overwrite: bool):
+    if access_method not in ["stream", "download", "local"]:
+        raise ValueError(
+            f"Invalid access method: {access_method}. Must be one of 'stream', 'download', 'local'"
+        )
+    if access_method in {"download", "local"}:
+        if not os.environ.get("HUB_DOWNLOAD_PATH"):
+            raise ValueError(
+                f"HUB_DOWNLOAD_PATH environment variable is not set. Cannot use access method '{access_method}'"
+            )
+        if overwrite:
+            raise ValueError(
+                "Cannot use access methods download or local with overwrite=True as these methods only interact with local copy of the dataset."
+            )
+
+
+def parse_access_method(access_method: str):
+    num_workers = None
+    scheduler = None
+    if access_method.startswith("download"):
+        split = access_method.split(":")
+        if len(split) == 1:
+            split.extend(("threaded", "0"))
+        elif len(split) == 2:
+            split.append("threaded" if split[1].isnumeric() else "0")
+        elif len(split) >= 3:
+            num_integers = sum(1 for i in split if i.isnumeric())
+            if num_integers != 1 or len(split) > 3:
+                raise ValueError(
+                    "Invalid access_method format. Expected format is one of the following: {download, download:scheduler, download:num_workers, download:scheduler:num_workers, download:num_workers:scheduler}"
+                )
+
+        access_method = "download"
+        num_worker_index = 1 if split[1].isnumeric() else 2
+        scheduler_index = 3 - num_worker_index
+        num_workers = int(split[num_worker_index])
+        scheduler = split[scheduler_index]
+    return access_method, num_workers, scheduler
+
+
+def get_local_dataset(
+    access_method,
+    path,
+    read_only,
+    memory_cache_size,
+    local_cache_size,
+    creds,
+    token,
+    verbose,
+    ds_exists,
+    num_workers,
+    scheduler,
+):
+    local_path = get_local_storage_path(path, os.environ["HUB_DOWNLOAD_PATH"])
+    if access_method == "download":
+        if not ds_exists:
+            raise DatasetHandlerError(
+                f"Dataset {path} does not exist. Cannot use access method 'download'"
+            )
+        elif hub.exists(local_path):
+            raise DatasetHandlerError(
+                f"A dataset already exists at the download location {local_path}. To reuse the dataset, use access method 'local'. If you want to download the dataset again, delete the dataset at the download location and try again."
```

Comment:
or use overwrite = True?

Reply:
Didn't go for this as it doesn't make it clear whether overwrite=True is overwriting the local dataset or cloud dataset.
Currently, we display this exception and show the path so the user can deliberately go and clear the cache.
Looping in @istranic

Reply:
Thought so, but haven't we made it clear that the cloud dataset cannot be modified in download/local mode?  Users might use it to update the dataset and feels like a hassle to delete manually.